python/hartree_fock/impurity.py

```diff
@@ -119,6 +147,34 @@ def f(Sigma_HF_flat):
                     if max_imag > 1e-10:
                         mpi.report('Warning! Discarding imaginary part of density matrix. Largest imaginary part: %f' % max_imag)
                     G_dens[bl] = G_dens[bl].real
+            
+            for bl, G0_bl in self.G0_iw:
+                #add DC
+                n_tot = G_iw.total_density().real
+                n_up = 0.5*n_tot
+                n_down = 0.5*n_tot
+                
+                if self.dc_fixed_occ is not None:
+                    mpi.report(f"\nHARTREE SOLVER: modifying occupations in DC calculation with given dc_fixed_occ = {self.dc_fixed_occ:.4f}")
+                    n_tot = self.dc_fixed_occ
+                    n_up = 0.5*n_tot
+                    n_down = 0.5*n_tot
+
+                if self.dc_fixed_value is None:
+                    if (self.dc_J != self.J) or (self.dc_U != self.U):
+                        mpi.report(f"\nHARTREE SOLVER: DC U and J were overwritten: ")
+                        mpi.report(f"U = {self.U:.4f} , dc_U = {self.dc_U:.4f} ")
+                        mpi.report(f"J = {self.J:.4f} , dc_J = {self.dc_J:.4f}\n")
+                    
+                    mpi.report(f"HARTREE SOLVER: Calling DC calculation:\n")
+                    DC_val, _ = compute_DC_from_density(n_tot, U = self.dc_U, J = self.dc_J, n_orbitals=self.n_orb, method=self.dc_type)
```

Comment:
this should be made optional. I understand that technically it makes almost no sense to run HF without this after our discussion, but in a model calculation someone may decide to not do it. For whatever reason.

Reply:
Removed U and J from solver call, now only dc_U and dc_J are passed, the solver defaults to no double counting and relies on user input to specify otherwise.

The initialization of the double counting parameters was moved to solid_dmft